### PR TITLE
Omg 334 don't stop validating exits after chain byzantine

### PR DESCRIPTION
**Pull request by purbanow**

### CODE REVIEW

File: apps/omg_api/lib/root_chain_coordinator.ex [hunk 1]
```diff
@@ -45,13 +45,17 @@ defmodule OMG.API.RootChainCoordinator do
...
 
     height_sync_interval = Application.fetch_env!(:omg_api, :ethereum_status_check_interval_ms)
     {:ok, _} = schedule_get_ethereum_height(height_sync_interval)
-    state = Core.init(allowed_services, rootchain_height)
-    request_sync(allowed_services)
+    state = Core.init(configs_services, rootchain_height)
+
+    configs_services
+    |> Map.keys()
+    |> request_sync()
```
Comment (pthomalla) [inline apps/omg_api/lib/root_chain_coordinator.ex]: what happened if send fails?

File: apps/omg_api/lib/root_chain_coordinator/core.ex [hunk 2]
```diff
@@ -89,8 +89,8 @@ defmodule OMG.API.RootChainCoordinator.Core do
     service.synced_height <= service_reported_sync_height and state.root_chain_height >= service_reported_sync_height
   end
 
-  defp get_services_to_sync(state, previous_synced_height) do
-    case get_synced_height(state) do
+  defp get_services_to_sync(state, service_name, previous_synced_height) do
+    case get_synced_height(state, service_name) do
```
Comment (pthomalla) [inline apps/omg_api/lib/root_chain_coordinator/core.ex]: here you send service_name? 
it should be pid

File: apps/omg_api/lib/root_chain_coordinator/core.ex [hunk 3]
```diff
@@ -119,13 +137,12 @@ defmodule OMG.API.RootChainCoordinator.Core do
...
-      state.services
-      |> Map.keys()
-      |> MapSet.new()
+  defp get_synced_height_by_mode(state, :sync_with_root_chain) do
+    {:sync, state.root_chain_height}
+  end
 
-    state.allowed_services == registered
+  defp all_services_checked_in?(state) do
+    state.configs_services |> Map.keys() == state.services |> Map.keys()
```
Comment (pthomalla) [inline apps/omg_api/lib/root_chain_coordinator/core.ex]: you assume that heys will be in this same order

File: apps/omg_api/lib/root_chain_coordinator/core.ex [hunk 4]
```diff
@@ -108,8 +108,26 @@ defmodule OMG.API.RootChainCoordinator.Core do
   @doc """
   Gets synchronized height
   """
-  @spec get_synced_height(t()) :: {:sync, non_neg_integer()} | :nosync
-  def get_synced_height(state) do
+  @spec get_synced_height(t(), atom() | pid()) :: {:sync, non_neg_integer()} | :nosync
+  def get_synced_height(state, pid) when is_pid(pid) do
```
Comment (pthomalla) [inline apps/omg_api/lib/root_chain_coordinator/core.ex]: what happens if one service will be restarts?

File: apps/omg_api/lib/application.ex [hunk 5]
```diff
@@ -29,7 +29,13 @@ defmodule OMG.API.Application do
       {OMG.API.BlockQueue.Server, []},
       {OMG.API.FreshBlocks, []},
       {OMG.API.FeeChecker, []},
-      {OMG.API.RootChainCoordinator, [:depositor, :exiter]},
+      {
+        OMG.API.RootChainCoordinator,
+        %{
+          :depositor => %{sync_mode: :sync_with_coordinator},
```
[the remark this answers is not in the document]
Comment (purbanow) [reply apps/omg_api/lib/application.ex]: In the case of `omg_watcher/lib/application`, I have:
```
  %{
          :depositor => %{sync_mode: :sync_with_coordinator},
          :exit_processor => %{sync_mode: :sync_with_root_chain},
          :exit_finalizer => %{sync_mode: :sync_with_coordinator},
          :exit_challenger => %{sync_mode: :sync_with_root_chain},
          OMG.Watcher.BlockGetter => %{sync_mode: :sync_with_coordinator}
        }
```
It is not possible to use `:` for `OMG.Watcher.BlockGetter` + it is not possible to mix `:` and `=>`

File: apps/omg_api/lib/root_chain_coordinator/core.ex (hunk 2 above)
Comment (purbanow) [reply apps/omg_api/lib/root_chain_coordinator/core.ex]: Nope, it should be `service_name`

File: apps/omg_api/lib/root_chain_coordinator/core.ex (hunk 3 above)
Comment (purbanow) [reply apps/omg_api/lib/root_chain_coordinator/core.ex]: Good catch!

File: apps/omg_api/lib/root_chain_coordinator/core.ex (hunk 4 above)
Comment (purbanow) [reply apps/omg_api/lib/root_chain_coordinator/core.ex]: We talk about, it should be fine as with every `check_in` call `pid` + `service_name` is passed.

File: apps/omg_api/lib/root_chain_coordinator.ex (hunk 1 above)
Comment (purbanow) [reply apps/omg_api/lib/root_chain_coordinator.ex]: So it fails, it's the same like before.

File: apps/omg_watcher/test/support/integration/test_helper.ex [hunk 6]
```diff
@@ -49,6 +49,10 @@ defmodule OMG.Watcher.Integration.TestHelper do
     decode16(data, ["txbytes", "proof", "sigs"])
   end
 
+  def assert_block_getter_down do
```
Comment (pthomalla) [inline apps/omg_watcher/test/support/integration/test_helper.ex]: change the name

File: apps/omg_api/lib/root_chain_coordinator/core.ex [hunk 7]
```diff
@@ -119,13 +137,13 @@ defmodule OMG.API.RootChainCoordinator.Core do
...
-      |> Map.keys()
-      |> MapSet.new()
+  defp get_synced_height_by_mode(state, :sync_with_root_chain) do
+    {:sync, state.root_chain_height}
+  end
 
-    state.allowed_services == registered
+  defp all_services_checked_in?(state) do
+    sort = fn map -> map |> Map.keys() |> Enum.sort() end
+    sort.(configs_services_keys) == sort.(services_keys)
```
Comment (pthomalla) [inline apps/omg_api/lib/root_chain_coordinator/core.ex]: where ```configs_services_keys``` is defined?

**Pull request merged**